vpn/src/main/java/com/duckduckgo/mobile/android/vpn/prefs/VpnSharedPreferencesProvider.kt

```diff
@@ -19,24 +19,75 @@ package com.duckduckgo.mobile.android.vpn.prefs
 import android.content.Context
 import android.content.Context.MODE_PRIVATE
 import android.content.SharedPreferences
+import androidx.core.content.edit
 import com.duckduckgo.di.scopes.AppScope
 import com.frybits.harmony.getHarmonySharedPreferences
 import com.squareup.anvil.annotations.ContributesBinding
+import timber.log.Timber
 import javax.inject.Inject
 
 interface VpnSharedPreferencesProvider {
-    fun getSharedPreferences(name: String, multiprocess: Boolean = false): SharedPreferences
+    /**
+     * Returns an instance of Shared preferences
+     * @param name Name of the shared preferences
+     * @param multiprocess `true` if the shared preferences will be accessed from several processes else `false`
+     * @param migrate `true` if the shared preferences existed prior to use the [VpnSharedPreferencesProvider], else `false`
+     */
+    fun getSharedPreferences(name: String, multiprocess: Boolean = false, migrate: Boolean = false): SharedPreferences
 }
 
+private const val MIGRATED_TO_HARMONY = "migrated_to_harmony"
+
 @ContributesBinding(AppScope::class)
 class VpnSharedPreferencesProviderImpl @Inject constructor(
     private val context: Context
 ) : VpnSharedPreferencesProvider {
-    override fun getSharedPreferences(name: String, multiprocess: Boolean): SharedPreferences {
+    override fun getSharedPreferences(name: String, multiprocess: Boolean, migrate: Boolean): SharedPreferences {
         return if (multiprocess) {
-            context.getHarmonySharedPreferences(name)
+            if (migrate) {
+                Timber.v("Migrate and return preferences to Harmony")
+                return migrateToHarmonyIfNecessary(name)
```

Comment:
`migrateToHarmonyIfNecessary` returns already the harmony prefs to save one call